File: model_obj/dimensions/dimDegree.m
classdef dimDegree < dimAngular
    %DIMDEGREE Dimension in degrees
    %   based off of the double class, this class implements required
    %   conversion methods.
     
   
    methods
        function new = toRadians(obj)
            %TORADIANS Convert this dimension to degrees
            
            new = dimRadian(obj*pi/180);
        end
        
        function new = toDegrees(obj) 
            %TODEGREES Convert this dimension to degrees
            
           new = obj; 
        end
    end
end

File: model_obj/dimensions/DimDegree.py
from DimAngular import DimAngular


class DimDegree(DimAngular):
    def __new__(cls, value):
        if type(value).__bases__[0] is DimAngular:
            value = value.to_degrees()
        return DimAngular.__new__(cls, value)

    def to_degrees(self):
        return self

    def to_radians(self):
        return self*3.14/180


    # def _to_dimes(self):
    #     return float(self)
    #
    # def _from_dimes(num1):
    #     x = num1
    #     return DimDegree(x)

        